src/patterns/grids/content-grid/content-grid.scss

```diff
@@ -0,0 +1,92 @@
+    $_secondary-column-count: floor((grid.$main-column-count - $_primary-column-count) / 2);
+    $_menu-column-count: grid.$main-column-count - $_primary-column-count - $_secondary-column-count;
+    --primary-column-inline-size: #{grid.get-overall-inline-size($_primary-column-count)};
+    --secondary-column-inline-size: #{get-overall-inline-size($_secondary-column-count)};
+    --menu-column-inline-size: #{get-overall-inline-size($_menu-column-count)};
+    grid-template-areas:
+      ". . menu    . ."
+      ". . primary . .";
+    grid-template-columns: [full-start] 1fr [main-start] var(--menu-column-inline-size) var(--primary-column-inline-size) var(--secondary-column-inline-size) [main-end] 1fr [full-end];
+  }
+
+  @include media-query.mq($from: extra-wide) {
+    $_primary-column-count: floor(grid.$main-column-count * 0.7);
+    $_secondary-column-count: floor((grid.$main-column-count - $_primary-column-count) / 2);
+    $_menu-column-count: grid.$main-column-count - $_primary-column-count - $_secondary-column-count;
+    --primary-column-inline-size: #{grid.get-overall-inline-size($_primary-column-count)};
+    --secondary-column-inline-size: #{grid.get-overall-inline-size($_secondary-column-count)};
+    --menu-column-inline-size: #{grid.get-overall-inline-size($_menu-column-count)};
+    grid-template-areas: ". menu primary . .";
+  }
+}
+
+.content-grid--has-secondary {
+  grid-template-areas:
+    ". menu      ."
+    ". primary   ."
+    ". secondary .";
+
+  @include media-query.mq($from: wide) {
+    $_primary-column-count: floor(grid.$main-column-count * 0.7);
+    $_secondary-column-count: grid.$main-column-count - $_primary-column-count;
+    --primary-column-inline-size: #{grid.get-overall-inline-size($_primary-column-count)};
+    --secondary-column-inline-size: #{grid.get-overall-inline-size($_secondary-column-count)};
+    grid-template-areas:
+      ". menu    menu      ."
+      ". primary secondary .";
+    grid-template-columns: [full-start] 1fr [main-start] var(--primary-column-inline-size) var(--secondary-column-inline-size) [main-end] 1fr [full-end];
+  }
+
+  @include media-query.mq($from: extra-wide) {
+    $_primary-column-count: floor(grid.$main-column-count * 0.6);
+    $_secondary-column-count: floor(grid.$main-column-count * 0.25);
+    $_menu-column-count: grid.$main-column-count - $_primary-column-count - $_secondary-column-count;
+    --primary-column-inline-size: #{grid.get-overall-inline-size($_primary-column-count)};
+    --secondary-column-inline-size: #{grid.get-overall-inline-size($_secondary-column-count)};
+    --menu-column-inline-size: #{grid.get-overall-inline-size($_menu-column-count)};
+    grid-template-areas: ". menu primary secondary .";
+    grid-template-columns: [full-start] 1fr [main-start] var(--menu-column-inline-size) var(--primary-column-inline-size) var(--secondary-column-inline-size) [main-end] 1fr [full-end];
+  }
+}
+
+.content-grid__item,
```

Comment:
Like line 10 are we still using BEM structure on classes etc?

Reply:
See https://github.com/libero/storybook/pull/39#discussion_r348496117